library/validator.py

```diff
@@ -73,72 +73,25 @@ class Validator:
     the library.
     """
 
-    def __init__(self, path):
+    def __init__(self, f):
+        self.__unparsed_file = f
 
-        # Abort if file path is not valid
-        if not self.__check_extension(path):
-            raise Exception("File path must point to a .yml or .yaml file")
-
-        self.path = path
-        self.fname = path.split("/")[-1].split(".")[0]
-
-    def __check_extension(self, path):
-        # Check if path ends with a .yml file
-        extension = path.split("/")[-1].split(".")[-1]
-        return extension in ["yml", "yaml"]
-
-    @cached_property
-    def __file(self):
-        with open(self.path, "r") as stream:
-            y = yaml.load(stream, Loader=yaml.FullLoader)
-            return y
+    def __call__(self):
+        assert self.tree_is_valid
+        assert self.dataset_name_matches
+        assert self.has_only_one_source
 
+    # Check that the tree structure fits the specified schema
     @property
     def tree_is_valid(self) -> bool:
-        if self.__file["dataset"] == None:
```

Comment:
looking good! now you can bring back the deleted logic so that they are not returning the default True